restx-core/src/main/java/restx/ResourcesRoute.java

```diff
@@ -93,6 +93,9 @@ public ResourcesRoute(String name, String baseRestPath, String baseResourcePath,
         this.baseRestPath = ("/" + checkNotNull(baseRestPath) + "/").replaceAll("/+", "/");
         this.baseResourcePath = checkNotNull(baseResourcePath)
                 .replace('.', '/').replaceAll("^/", "").replaceAll("/$", "") + "/";
+        if("/".equals(this.baseResourcePath)){
```

Comment:
maybe we could allow to do that with a special setting. In some situations, security is not an issue (if the server is isolated from the open world). Maybe something like `restx.security.allowServingRootResources`? WDYT?

Reply:
Agree on the possibility to disable this security check.

To proceed, we would need to retrieve `SecuritySettings` :
- Either by passing `SecuritySettings` to `ResourcesRoute`'s constructor (which would be another breaking change) after having injected in subclasses
- Or by retrieving it statically through `Factory

After thinking a bit about it, I'm not sure if a "configuration wide" property is adequate here, as we're *obviously not* using `/` baseResourcePath in classes relying on `ResourcesRoute` today in RestX source (and I expect this will never be the case in the case either).

Which means we will only have 1 `ResourcesRoute` which would be able to be mounted on `/` baseRetourcePath AND this will be a developer-defined `ResourcesRoute`.
=> I assume only providing a `forceServingResourcesFromRootClasspath` constructor flag (obviously defaulted to `false`) would be enough.
I'm even considering having an overridable (`protected`) `ensureBaseResourcePathValid()` :
```
    public ResourcesRoute(String name, String baseRestPath, String baseResourcePath, ImmutableMap<String, String> aliases, List<CachedResourcePolicy> cachedResourcePolicies) {
        this.name = checkNotNull(name);
        this.baseRestPath = ("/" + checkNotNull(baseRestPath) + "/").replaceAll("/+", "/");
        this.baseResourcePath = this.ensureBaseResourcePathValid(baseResourcePath);
        this.aliases = checkNotNull(aliases);
        this.cachedResourcePolicies = ImmutableList.copyOf(cachedResourcePolicies);
    }

    protected String ensureBaseResourcePathValid(String baseResourcePath) {
        String escapedBaseResourcePath = checkNotNull(baseResourcePath)
                .replace('.', '/')
                .replaceAll("^/", "")
                .replaceAll("/$", "") + "/";

        if("/".equals(escapedBaseResourcePath)){
            throw new IllegalArgumentException("Please, avoid using '/' as ResourcesRoute's baseResourcePath as it represents serious security flaws (people will be able to read your classpath configuration files)");
        }
        
        return escapedBaseResourcePath;
    }
```

The day someone will want to serve resources from `/` in the classpath, he would have to explicitely override this method in his `ResourcesRoute` implementation (doing ... nothing)

WDYT ?